runbotics/runbotics-common/src/Expressions/index.ts

```diff
@@ -68,21 +70,61 @@ export class Expressions {
             context?.environment?.output?.variableName;
     }
 
-    private static getFullServiceMethodCall(templatedString, context, expressionFnContext) {
-        const extractedMethod = jexlServicePattern.exec(templatedString);
-        const methodName = extractedMethod[1];
-        const args = extractedMethod[2].trim().split(",");
-
-        const mappedArgs = args.map((arg) => {
-            const trimmedArg = arg.trim();
-            if (jexlServicePattern.test(trimmedArg)) {
-                return Expressions.resolveExpression(trimmedArg, context, expressionFnContext);
-            } else if (Expressions.isVariableArgument(trimmedArg)) {
-                return 'environment.variables.' + trimmedArg;
+    private static getFullServiceMethodCall(templatedString: string, context, expressionFnContext) {
+        const jexlRecursiveExp = new RegExp(jexlRecursivePattern, 'g');
+        const matches = templatedString.match(jexlRecursiveExp);
+        if (!matches) {
+            return templatedString;
+        }
+
+        let result = templatedString;
+        for (const expressionMatch of matches) {
+            const recursivelyExtractedExpression =
+                jexlRecursiveExp.exec(templatedString);
+            if (!recursivelyExtractedExpression) continue;
+
+            const extractedMethod =
+                jexlServiceMethodPattern.exec(recursivelyExtractedExpression[1]);
+            if (!extractedMethod) continue;
+
+            const methodName = extractedMethod[1];
+            const args = extractedMethod[2].length
+                ? extractedMethod[2].trim().split(",")
```

Comment:
create variable for extractedMethod[2] - the code will be more readable
can be done also with:
const [methodName, extractedMethodSecond] = extractedMethod;